### DISCUSSION TITLE
Does Submariner detect and attempt to squash duplicate tracks between servers?

### DISCUSSION THREAD

**Discussion by simensen** [SubmarinerApp/Submariner#133] [Q&A]
Hello,

New to Submariner, here! I've finally set up my Navidrome servers the way I want and finding some weirdness with some of the tracks when navigating with Submariner.

My setup:

```
# Filesystem
/srv/Music
  DJ/       // tracks that are specifically intended for djing 
  Mixes/    // my own DJ mixes and downloads/recordings from other DJs
  Music/    // commercial music, albums, etc.

# Navidrome.lan – instance indexes all music under /srv/Music
/srv/Music:/music

# Navidrome-DJ.lan – instance indexes just the DJ library
/srv/Music/DJ:/music

# Navidrome-Mixes.lan – instance indexes just the Mixes library
/srv/Music/Mixes:/music

# Navidrome-Music.lan – instance indexes just the Music library
/srv/Music/Music:/music
```

When I visit *navidrome.lan*, *navidrome-dj.lan*, *navidrome-mixes*, or *navidrome-music.lan* via browser, I see what I expect to see. The big *navidrome.lan* instance shows me everything across all three directories. When I visit any of the other three, I only see songs for their specific directories.

However, when I add all four instances as separate servers to Submariner, things are... a bit weird. It *looks* like Submariner is looking at the four libraries and only including a specific track *once*. So, since I loaded up *navidrome-dj.lan* first, the *albums* show up under `navidrome.lan` (including some album art!), but the actual list of tracks for those albums are empty and have no tracks at all.

To experiment, I removed *navidrome-dj.lan*, *navidrome-mixes*, or *navidrome-music.lan* as remote servers and left *navidrome.lan*. Once I did that, all the missing tracks appeared under the albums as expected.

I'd love to have the option to keep all four instances running. In general, the main instance (*navidrome.lan*) is probably what I'll use, but I'd love to be able to select one of the other two instances as needed if I want to filter based on supertype.

Is this a bug or working as intended?

**Comment by NattyNarwhal** [comment]
The artists/albums/tracks for each should be separate for each server. I wouldn't be surprised if there might be an issue with missing server predicates when doing queries though.

FWIW Submariner also doesn't support the "music folder" concept in the API, but I don't think Navidrome supports that.

**Comment by NattyNarwhal** [comment]
I did look at the track fetch code in c31087a. I wonder if that might help.

**Reply by simensen** [reply]
What is the best way for me to try c31087a? I've never built a project like this by hand, so that could be fun. :)

**Reply by NattyNarwhal** [reply]
You can try the build artifacts from CI runs, i.e. here: https://github.com/NattyNarwhal/Submariner/actions/runs/5159606089 - There aren't signed, however.

If you want to try building it yourself, install Xcode and try the instructions here: https://github.com/NattyNarwhal/Submariner#building
